### PR TITLE
[prometheus-node-exporter] Add Termination Msg Params to Daemonset

### PR DESCRIPTION
**Pull request by nfrazier08**
#### What this PR does / why we need it

For our purposes of this chart, we would like to be able to customize the node-exporter daemonset with the following variables:

- For both the Node Exporter and KubeRBACProxy containers, variables have been added to configure `terminationMessagePath` and `terminationMessagePolicy`
- A variable was added to configure the `terminationGracePeriodSeconds`

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

@zeritti @gianrubio @zanhsieh

### CODE REVIEW

**Review by zanhsieh — CHANGES_REQUESTED** [review]
@nfrazier08 
Can you separate feature branch and bug fix? You can choose patching the bug go first or the feature go first, but not both.

**Comment by nfrazier08** [comment]
@zanhsieh Thanks for taking a look at this PR. I will do the bug fix in a following PR and have removed it from this PR.

File: charts/prometheus-node-exporter/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ keywords:
   - prometheus
   - exporter
 type: application
-version: 4.26.1
+version: 4.27.1
```
Comment (zanhsieh) [inline charts/prometheus-node-exporter/Chart.yaml:9]:
```suggestion
version: 4.27.0
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zeritti — CHANGES_REQUESTED** [review]
@nfrazier08 Thank you for your PR. This PR, though, covers too many different and unrelated changes on:
- termination params
- volume mounts
- restart policy
- support for environment variables
- container ports

Please, prefer related changes in a PR, ideally on a single topic/area. It is reasonable, indeed, to raise more than one PR on these changes.

**Comment by nfrazier08** [comment]
@zeritti - Thank you, I will go ahead and split these changes up into separate PRs - starting with this one for adding termination params.

**Review by zeritti — CHANGES_REQUESTED** [review]
When seeing a set of related parameters with a toggle, the first thing to occur to me is a map: the toggle if set enables all the fields in the map. An example of a map that one might name, say, `terminationMessageParams` in the values file:

```yaml
## values.yaml
# Enable or disable container termination message settings
# https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/
terminationMessageParams:
  enabled: false
  # If enabled, specify the path for termination messages
  terminationMessagePath: /dev/termination-log
  # If enabled, specify the Policy for termination messages
  terminationMessagePolicy: File
```

and in the template (example for the main container):

```yaml
## template
{{- if .Values.terminationMessageParams.enabled }}
{{- with .Values.terminationMessageParams }}
terminationMessagePath: {{ .terminationMessagePath }}
terminationMessagePolicy: {{ .terminationMessagePolicy }}
{{- end }}
{{- end }}
```

I have not yet seen a case where containers in the same pod are using different termination message parameters. If this indeed occurs as you foresee, the code has to be applied per container in both the values and the template like you show. Otherwise, it can be set in the values just once and shared across the two containers. 

Using `configureTerminationMessagePath` as a toggle may mislead as it does not configure just `terminationMessagePath`.

Alternatively, without a toggle:

```yaml
# Specify the path for termination messages
terminationMessagePath: ""
# Specify the Policy for termination messages
terminationMessagePolicy: ""
```

```yaml
{{- if .Values.terminationMessagePath }}
terminationMessagePath: {{ .Values.terminationMessagePath }}
{{- end }}
{{- if .Values.terminationMessagePolicy }}
terminationMessagePolicy: {{ .Values.terminationMessagePolicy }}
{{- end }}
```

**Comment by nfrazier08** [comment]
@zeritti You are right, I like the idea of a toggle better and I don't think the terminationMessagePath or terminationMessagePolicy are going to be different for each container.

**Review by zeritti — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/values.yaml [hunk 2]
```diff
@@ -361,10 +361,23 @@ nodeSelector:
   kubernetes.io/os: linux
   #  kubernetes.io/arch: amd64
 
+# Specify grace period for graceful termination of pods. Defaults to 30
```
Comment (zanhsieh) [inline charts/prometheus-node-exporter/values.yaml:364]:
```suggestion
# Specify grace period for graceful termination of pods. Defaults to 30 if null or not specified
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/values.yaml (hunk 2 above)
Comment (nfrazier08) [reply charts/prometheus-node-exporter/values.yaml]: Changed.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**